Support Trait Image Upload, Crop Ontology and Other Improvements

reynoldtan opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!--- No need to include the issue number in the title :-) -->
- This PR will improve data importer to support image upload and crop ontology association.
- Fixed theme breaking when validation error encountered.
- Set the button in last stage to Save.
- Permit use of parenthesis in trait unit.
- Fixed empty summary table when quotes are present in Stage 3 - Describe trait.
- Fixed errors caused by quotes.
## Metadata
<!--- If it fixes an open issue, please add the issue link below. -->
 - Issue  https://github.com/UofS-Pulse-Binfo/analyzedphenotypes/issues/101 Improve Phenotype Importer

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] I feel this PR is ready to be merged.
- [ ] This PR is dependent upon [PR #/ nothing]

Documentation:
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

## Description
<!--- Describe your changes in detail -->
<!--- Why is this change required? What problem does it solve? -->

## Dependencies
<!-- If this code is dependent upon another module and/or PR, 
       state that here. -->
<!-- Include information about other modules/PRs needed for testing,
        which to enable/merge first, etc. -->

## Testing?
Tested:
1. Upload image to a new trait: 
  a. both image uploaded
  b. one image uploaded
2. Reusing trait:
  a. both image retrieved
  b. single image retrieved 
  c. no image - 'Photo not available'.
3. Ontology terms suggested based on words in a trait name.
  a. Ontology term entered that does not exist.
4. Upload data
  a. Select an ontology term
  b. No ontology term selected
5. Reusing trait
  a. Ontology field prefilled with previously set term.
6. Parenthesis in unit
7. Quotes in all column headers.

Note:
1. Instruction that pertains to column headers does not include 'Value' which, following the order of header, should be #9 in the list and data collector being the last.
2. File upload does not support spaces in the file name.

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how
      your change affects other areas of the code, etc. -->
<!--- Reviewers will use this section to test the submission! -->
- [X] I tested on a generic Tripal Site
- [ ] I tested on a KnowPulse Clone
- [ ] This PR includes automated testing

reynoldtan committed:
Support Trait Image Upload

reynoldtan committed:
Support for image and ontology for review

reynoldtan committed:
Set the button in last stage to save

reynoldtan committed:
permit use of parethesis in trait unit

reynoldtan committed:
Fixed empty summary table when quotes are present in Stage 3 - Describe Trait

reynoldtan committed:
Update APPhenotypeImporter.inc

reynoldtan committed:
Improve data importer

reynoldtan committed:
Merge branch 'Phenotype-Image-Support' of https://github.com/UofS-Pulse-Binfo/analyzedphenotypes into Phenotype-Image-Support

reynoldtan committed:
support data import using server/local file path

laceysanderson reviewed (changes requested):
I receive a lot of errors when trying to use the phenotype importer on this branch. Specifically,
![Screen Shot 2019-10-22 at 9 52 41 AM](https://user-images.githubusercontent.com/1566301/67306366-14b1d480-f4b4-11e9-90e6-b9d4c7f81ccc.png)
Then after submission of the describe stage I see this.
![Screen Shot 2019-10-22 at 9 55 07 AM](https://user-images.githubusercontent.com/1566301/67306375-17acc500-f4b4-11e9-9c61-023e7c6a0fd0.png)

I also receive the following errors on the trait page when trying to confirm the image was saved properly.
![Screen Shot 2019-10-22 at 10 00 03 AM](https://user-images.githubusercontent.com/1566301/67306378-1a0f1f00-f4b4-11e9-8f70-0bcdcdadc1cf.png)

I have confirmed the local file element works though! 🙃

reynoldtan committed:
Bug fixes

Fixes bugs reported via Drupal logging and errors, including undefined
variables and non-numeric value.

reynoldtan committed:
Fixed duplicate key error.

Added a check to make sure that a key combination does not exist before
adding a row into collections table.

reynoldtan committed:
Updated instruction

Included Value column in the instructions list

laceysanderson committed:
Small fixes to ontology terms.

laceysanderson reviewed (approved):
Works great!
